### PR TITLE
Improve error message for time types

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2376

Fixes:

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java [hunk 1]
```diff
@@ -135,7 +136,11 @@ private static Object nativeExecuteExecutable(BObject client, BObject paramSQLSt
             } catch (ApplicationError e) {
                 return ErrorGenerator.getSQLApplicationError(e);
             } catch (Throwable th) {
-                return ErrorGenerator.getSQLError(th, String.format("Error while executing SQL query: %s. ", sqlQuery));
+                if (th instanceof BError) {
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java:139]: We can catch BError is I am not mistaken.

File: native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultStatementParameterProcessor.java [hunk 2]
```diff
@@ -480,7 +493,8 @@ public static DefaultStatementParameterProcessor getInstance() {
                             java.time.format.DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
                     arrayData[i] = LocalDateTime.parse(innerValue.toString(), formatter);
                 } catch (java.time.format.DateTimeParseException ex) {
-                    throw new DataError("Unsupported String Value " + innerValue + " for DateTime Array");
+                    throw ErrorGenerator.getConversionError("Unsupported string value " + innerValue +
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultStatementParameterProcessor.java:496]:
```suggestion
                    throw new ConversionError("Unsupported string value " + innerValue +
```

File: native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultStatementParameterProcessor.java [hunk 3]
```diff
@@ -316,54 +323,60 @@ public static DefaultStatementParameterProcessor getInstance() {
...
+                        int intZoneSecond = zoneSeconds.decimalValue().setScale(0, RoundingMode.HALF_UP)
+                                .intValue();
+                        OffsetTime offsetTime = OffsetTime.of(localTime,
+                                ZoneOffset.ofHoursMinutesSeconds(zoneHours, zoneMinutes, intZoneSecond));
+                        arrayData[i] = offsetTime;
+                    } else {
+                        arrayData[i] = Time.valueOf(localTime);
+                    }
+                } catch (Throwable e) {
+                    throw ErrorGenerator.getConversionError("Unsupported value: " + innerValue +
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultStatementParameterProcessor.java:378]: You can simply throw java exception, this should be handled at the higher level

File: native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java (hunk 1 above)
Comment (kalaiyarasiganeshalingam) [reply native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/ExecuteProcessor.java]: Removed this as no need this check when throwing an exception.

File: native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultStatementParameterProcessor.java [hunk 4]
```diff
@@ -1142,16 +1162,24 @@ protected void setDate(PreparedStatement preparedStatement, String sqlType, int
...
+                    int year = Math.toIntExact(dateMap.getIntValue(StringUtils.
+                            fromString(io.ballerina.stdlib.time.util.Constants.DATE_RECORD_YEAR)));
+                    int month = Math.toIntExact(dateMap.getIntValue(StringUtils.
+                            fromString(io.ballerina.stdlib.time.util.Constants.DATE_RECORD_MONTH)));
+                    int day = Math.toIntExact(dateMap.getIntValue(StringUtils.
+                            fromString(io.ballerina.stdlib.time.util.Constants.DATE_RECORD_DAY)));
+                    date = Date.valueOf(year + "-" + month + "-" + day);
+                } catch (Throwable e) {
+                    throw new ConversionError("Unsupported value: " + value + " for Date Value");
+                }
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultStatementParameterProcessor.java:1182]: Lets move the entire if and else inside a try catch

**Review by niveathika — APPROVED** [review]

**Pull request merged**